### PR TITLE
refactored ChainDataProvider tests

### PR DESCRIPTION
**Pull request by rolfyone**
These tests were supposed to be clear and easy to read and there's so many permutations of things now they're defeating the purpose. Hopefully this reduces some of the noise.

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  @BeforeEach
+  public void setup() {
+    spec = getSpec();
+    this.data = new DataStructureUtil(spec);
+    specConfig = spec.getGenesisSpecConfig();
+    storageSystem = InMemoryStorageSystemBuilder.buildDefault(StateStorageMode.ARCHIVE, spec);
+    SpecConfig specConfig = spec.getGenesisSpecConfig();
+
+    final UInt64 slot = UInt64.valueOf(specConfig.getSlotsPerEpoch() * 3L);
```
Comment (lucassaldanha) [inline data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java:72]: Where does `3L` come from?

File: data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+  protected abstract Spec getSpec();
+
+  @BeforeEach
+  public void setup() {
+    spec = getSpec();
+    this.data = new DataStructureUtil(spec);
+    specConfig = spec.getGenesisSpecConfig();
+    storageSystem = InMemoryStorageSystemBuilder.buildDefault(StateStorageMode.ARCHIVE, spec);
+    SpecConfig specConfig = spec.getGenesisSpecConfig();
```
[the remark this answers is not in the document]
Comment (rolfyone) [reply data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java]: actually a fair point, fixed.

**Review by lucassaldanha — COMMENTED** [review]
LGTM

**Review by courtneyeh — APPROVED** [review]
LGTM

File: data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java (hunk 1 above)
Comment (rolfyone) [reply data/provider/src/test/java/tech/pegasys/teku/api/AbstractChainDataProviderTest.java:72]: just moving code, didnt make this one up

**Pull request merged**